### ISSUE TITLE
chpldoc outputs incorrect documentation for an initialization usecase

### ISSUE THREAD

**Issue by prashanth018** [chapel-lang/chapel#17110]
### Summary of Problem
Global variable with the below initialization expression don't print out reasonably in the generated docs - string literal arguments lose their quotation marks in the generated documentation.

### Steps to Reproduce

**Source Code:**

```chapel
var a = foo("this is", a, "function", call);
```

**produces output like:**

```
.. default-domain:: chpl

.. module:: variablepatterns.doc

variablepatterns.doc
====================
**Usage**

.. code-block:: chapel

   use variablepatterns.doc;


or

.. code-block:: chapel

   import variablepatterns.doc;

.. data:: var a = foo(this is, a, function, call)
```

**Expected output:**
`var a = foo("this is", a, "function", call)`

**Compile command:**
`chpldoc filename.chpl`

**Execution command:**
visit the generated web page

**Associated Future Test(s):**
[`test/chpldoc/globals/variablepatterns.doc.chpl`](
           https://github.com/chapel-lang/chapel/blob/master/test/chpldoc/globals/variablepatterns.doc.chpl
          ) #17111

**Comment by lydia-duncan** [comment]
Note that I edited the issue to be a little more specific about what is wrong.  Thanks for filing :)

**Comment by tt-leader** [comment]
Hi, Lydia, I would like to work on this issue, and I have talked about my ideas on [Gitter GSoC channel](https://gitter.im/chapel-lang/chapel-summer-of-code)

**Comment by lydia-duncan** [comment]
Resolved by #17459
